lib/sles4sap_publiccloud.pm

```diff
@@ -999,6 +1000,91 @@ sub saphanasr_showAttr_version {
     }
 }
 
+=head2 wait_for_cluster
+
+    Verifies that nodes are online, resources are started and DB is in sync
+
+=cut
+
+sub wait_for_cluster {
+    my ($self, $wait_time, $max_retries) = @_;
+
+    $wait_time //= 10;
+    $max_retries //= 7;
+
+    while ($max_retries > 0) {
+        my $hanasr_output = $self->run_cmd(cmd => 'SAPHanaSR-showAttr', quiet => 1);
+        my $crm_output = $self->run_cmd(cmd => 'crm_mon -r -R -n -N -1', quiet => 1);
+
+        my @lines = split /\n/, $hanasr_output;
+
+        # Look for the line that starts with "Hosts"
+        my $start_idx = 0;
+        for (my $i = 0; $i < @lines; $i++) {
+            if ($lines[$i] =~ /^\s*Hosts\s+/) {
+                $start_idx = $i;
+                last;
+            }
+        }
+
+        # Only get lines after here
+        @lines = @lines[$start_idx..$#lines];
+
+        # Get rif of header
+        shift @lines;
+
+        my $found_prim = 0;
+        my $found_sok = 0;
+        my $is_online = 0;
+        my $resource_started = 1;
+        my $failed_actions = 0;
+
+        # Check for resources that are still starting
+        if ($crm_output =~ /:\s*Starting/) {
+            $resource_started = 0;
+        }
+
+        # Check for Failed resource actions
+        if ($crm_output =~ /Failed Resource Actions:/) {
+            $failed_actions = 1;
+        }
+
+        # Check nodes and sync state
+        foreach my $line (@lines) {
+            next if $line =~ /^-+/; # Skip any dashed lines between headers and data
+            my @fields = split ' ', $line;
+
+            foreach my $field (@fields) {
+                $is_online += 1 if $field eq 'online';
+                $found_prim = 1 if $field eq 'PRIM';
+                $found_sok = 1 if $field eq 'SOK';
```

Comment:
I tried that on the non-script command and wasn't doable, because some fields have a value in one line but not on the second, leading to whitespace-splitting creating different indexes for the 2 VM Lines. I'll try with `script` and see if that's solved. 

Example: http://openqaworker15.qa.suse.cz/tests/277916#step/Crash_replica/44 see the column next to `site_a` and `site_b`. `vmhana01` has a `-` that's missing from vmhana02.